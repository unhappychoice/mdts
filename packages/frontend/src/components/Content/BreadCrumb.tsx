import { Breadcrumbs, Link, Typography } from '@mui/material';
import React from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '../../store/store';

interface BreadCrumbProps {
  onDirectorySelect?: (directoryPath: string) => void;
}

const BreadCrumb: React.FC<BreadCrumbProps> = ({ onDirectorySelect }) => {
  const { currentPath } = useSelector((state: RootState) => state.history);

  const pathSegments = currentPath ? currentPath.split('/').filter(segment => segment !== '') : [];

  return (
    <Breadcrumbs aria-label="breadcrumb" sx={{ mb: 4 }}>
      {pathSegments.map((segment, index) => {
        const isLast = index === pathSegments.length - 1;
        const path = pathSegments.slice(0, index + 1).join('/');
        return isLast ? (
          <Typography key={path} color="text.primary">
            {segment}
          </Typography>
        ) : (
          <Link key={path} color="inherit" href="#" onClick={() => onDirectorySelect && onDirectorySelect(path)}>
            {segment}
          </Link>
        );
      })}
    </Breadcrumbs>
  );
};

export default BreadCrumb;
