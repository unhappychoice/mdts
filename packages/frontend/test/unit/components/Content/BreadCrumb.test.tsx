import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import BreadCrumb from '../../../../src/components/Content/BreadCrumb';

const mockStore = configureStore([]);

describe('BreadCrumb', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      history: {
        currentPath: '',
      },
    });
  });

  test('renders nothing when currentPath is empty', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BreadCrumb />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders breadcrumbs for a file path', () => {
    store = mockStore({
      history: {
        currentPath: 'folder1/subfolder/file.md',
      },
    });
    const { asFragment } = render(
      <Provider store={store}>
        <BreadCrumb />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('folder1')).toBeInTheDocument();
    expect(screen.getByText('subfolder')).toBeInTheDocument();
    expect(screen.getByText('file.md')).toBeInTheDocument();
  });

  test('renders breadcrumbs for a directory path', () => {
    store = mockStore({
      history: {
        currentPath: 'folder1/subfolder',
      },
    });
    const { asFragment } = render(
      <Provider store={store}>
        <BreadCrumb />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('folder1')).toBeInTheDocument();
    expect(screen.getByText('subfolder')).toBeInTheDocument();
  });

  test('calls onDirectorySelect when a directory link is clicked', () => {
    const onDirectorySelectMock = jest.fn();
    store = mockStore({
      history: {
        currentPath: 'folder1/subfolder/file.md',
      },
    });
    render(
      <Provider store={store}>
        <BreadCrumb onDirectorySelect={onDirectorySelectMock} />
      </Provider>
    );

    fireEvent.click(screen.getByText('folder1'));
    expect(onDirectorySelectMock).toHaveBeenCalledWith('folder1');

    fireEvent.click(screen.getByText('subfolder'));
    expect(onDirectorySelectMock).toHaveBeenCalledWith('folder1/subfolder');
  });
});
