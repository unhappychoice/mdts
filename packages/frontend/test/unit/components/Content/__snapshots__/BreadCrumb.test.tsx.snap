// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BreadCrumb renders breadcrumbs for a directory path 1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1eq4jhp-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-i9kmoy-MuiTypography-root-MuiLink-root"
          href="#"
        >
          folder1
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-14ja1gv-MuiTypography-root"
        >
          subfolder
        </p>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;

exports[`BreadCrumb renders breadcrumbs for a file path 1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1eq4jhp-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-i9kmoy-MuiTypography-root-MuiLink-root"
          href="#"
        >
          folder1
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-i9kmoy-MuiTypography-root-MuiLink-root"
          href="#"
        >
          subfolder
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-14ja1gv-MuiTypography-root"
        >
          file.md
        </p>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;

exports[`BreadCrumb renders nothing when currentPath is empty 1`] = `
<DocumentFragment>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1eq4jhp-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
    />
  </nav>
</DocumentFragment>
`;
